index.js

```diff
@@ -167,6 +175,30 @@ program
       .readSiteConfig()
       .then((config) => {
         serverConfig.mount.push([config.baseUrl || '/', outputFolder]);
+
+        if (onePagePath) {
+          const lazyReloadMiddleware = function (req, res, next) {
+            const isHtmlFile = req.url.endsWith('.html');
+
+            if (isHtmlFile) {
+              const isDynamicIncludeHtmlFile = req.url.endsWith('._include_.html');
+
+              if (!isDynamicIncludeHtmlFile) {
+                const urlWithoutBaseUrl = req.url.replace(config.baseUrl, '');
+                const urlWithoutExtension = fsUtil.removeExtension(urlWithoutBaseUrl);
+
+                const didInitiateRebuild = site.changeCurrentPage(urlWithoutExtension);
+                if (didInitiateRebuild) {
+                  req.url = ensurePosix(path.join(config.baseUrl || '/', LAZY_LOADING_SITE_FILE_NAME));
+                }
+              }
+            }
+            next();
+          };
+
+          serverConfig.middleware = [lazyReloadMiddleware];
```

Comment:
Initialize an empty array for `middleware` during the initialization of `serverConfig`, then replace this with a push instead.

Reply:
Updated; Thanks for looking through this!